pages/index.tsx

```diff
@@ -110,7 +115,47 @@ const HomePage = (props: any) => {
     props.editMode
   )
 
+  const [authPopupDisplayed, setAuthPopupDisplayed] = useState(false)
+
+  const refreshPage = () => {
+    fetch(`/api/reset-preview`).then(() => {
+      window.location.href = "/?autoAuth"
+    })
+  }
+
+  useEffect(() => {
+    if (window.location.href.includes("autoAuth")) {
+      setAuthPopupDisplayed(true)
+    }
+  })
+
+  
+  
+
   return (
+    <InlineForm form={form}>
+      {authPopupDisplayed && (
+        <ActionableModal title="Authentication" message="To edit this site, you first need to be authenticated." actions={[
+          {
+            name: "Continue",
+            action: enterEditMode,
```

Comment:
We may want to remove the `autoAuth` query param on load, so that if they accept or cancel, it won't re-pop upon refresh (or on page change)